usr/src/uts/common/fs/zfs/dmu_send.c

```diff
@@ -2669,6 +2678,18 @@ receive_read_record(struct receive_arg *ra)
 	case DRR_WRITE:
 	{
 		struct drr_write *drrw = &ra->rrd->header.drr_u.drr_write;
+		if (drrw->drr_compressiontype >= ZIO_COMPRESS_FUNCTIONS)
+			return (SET_ERROR(ENOTSUP));
+		if (drrw->drr_checksumtype >= ZIO_CHECKSUM_FUNCTIONS)
+			return (SET_ERROR(ENOTSUP));
+		if (P2PHASE(drrw->drrw->drr_logical_size, SPA_MINBLOCKSIZE))
```

Comment:
This seems like both a typo and a duplication of line 2691?

Reply:
fixed